modules/smtp/strong_password/files/smtp_honeypot.py

```diff
@@ -0,0 +1,346 @@
+    output_lock.acquire()
+    try:
+        logfile_handle = open(LOGFILE, "a")
+        logfile_handle.write(
+            json.dumps(
+                {
+                    "ip": ip,
+                    "data": data,
+                    "port": port,
+                    "module_name": "smtp/mail_honeypot",
+                    'date': datetime.datetime.now().\
+                    strftime("%Y-%m-%d %H:%M:%S")
+                }
+            ) + "\n"
+        )
+        logfile_handle.close()
+    finally:
+        output_lock.release()
+    #print(file_path,ip,port,data)
+    #with output_lock:
+        #with open(file_path, "a") as f:
+    message = "[{0}][{1}:{2}] {3}".format(time.time(), ip, port, data)
+    #print(message)
+
+
+
+class SMTPChannel(asynchat.async_chat):
+    COMMAND = 0
+    DATA =1
+
+    def __init__(self, server, conn, addr):
+        asynchat.async_chat.__init__(self, conn)
+        self.__rolling_buffer = ""
+        self.__server = server
+        self.__conn = conn
+        self.__addr = addr
+        self.__line = []
+        self.__state = self.COMMAND
+        self.__greeting = 0
+        self.__mailfrom = None
+        self.__rcpttos = []
+        self.__data = ''
+        self.__fqdn = MAILSERVER_NAME
+        try:
+            self.__peer = conn.getpeername()
+        except socket.error as err:
+            # a race condition  may occur if the other end is closing
+            # before we can get the peername
+            self.close()
+            if err[0] != errno.ENOTCONN:
+                raise
+            return
+        #print >> DEBUGSTREAM, 'Peer:', repr(self.__peer)
+        self.push('220 %s %s' % (self.__fqdn, __version__))
+        self.set_terminator(b'\r\n')
+
+    # Overrides base class for convenience
+    def push(self, msg):
+        msg = msg + '\r\n'
+        msg = msg.encode()
+        asynchat.async_chat.push(self, msg)
+
+    # # Implementation of base class abstract method
+    def collect_incoming_data(self, data):
+        data = data.decode('utf-8')
+        self.__line.append(data)
+        del data
+
+    # Implementation of base class abstract method
+    def found_terminator(self):
+        line = EMPTYSTRING.join(self.__line)
+        log_to_file(self.__addr[0], self.__addr[1], line)
+
+        self.__line = []
+        if self.__state == self.COMMAND:
+            if not line:
+                self.push('500 Error: bad syntax')
+                return
+            method = None
+            i = line.find(' ')
+            if i < 0:
+                command = line.upper()
+                arg = None
+            else:
+                command = line[:i].upper()
+                arg = line[i+1:].strip()
+            #command = command.decode('utf-8')
+            method = getattr(self, 'smtp_' + command, None)
+            if not method:
+                self.push('502 Error: command "%s" not implemented' % command)
+                return
+            method(arg)
```

Comment:
If the `method` function is set as `None` by `getattr` this will give an error. Check if the value is not `None` before calling the function.

Reply:
@ChakshuGupta : it's done exactly the line above.

Reply:
Sorry my bad!